src/core/controller.ts

```diff
@@ -190,12 +196,36 @@ export class Controller {
     return this.bot.store.get(`executors.${rule.name}.currentIdx`) || 0;
   }
 
-  setRuleExecutorIndex(rule: Rule, value?: any): void {
-    if (value !== undefined) {
-      this.bot.store.set(`executors.${rule.name}.currentIdx`, value);
-    } else {
-      this.bot.store.unset(`executors.${rule.name}.currentIdx`);
+  incRuleExecutorIndex(rule: Rule): void {
+    this.bot.store.set(
+      `executors.${rule.name}.currentIdx`, this.getRuleExecutorIndex(rule) + 1
+    );
+  }
+
+  resetRuleExecutorIndex(rule: Rule): void {
+    this.bot.store.unset(`executors.${rule.name}.currentIdx`);
+  }
+
+  async executeRuleTypeExecutors(rule: Rule, lastAnswer: Answer | Answer[]): Promise<any> {
```

Comment:
`Promise<Answer | Answer[]>` breaks?

Reply:
There is no guarantee that return will be answer... the return will be whatever transform + validators returns

Reply:
done